renovate.json

```diff
@@ -0,0 +1,32 @@
+{
+    "$schema": "https://docs.renovatebot.com/renovate-schema.json",
+    "extends": [
+      "config:base"
+    ],
+    "prHourlyLimit": 4,
+    "baseBranches": ["main", "release-2.10", "release-2.9"],
+    "packageRules": [
+      {
+        "matchBaseBranches": ["release-2.10","release-2.9"],
+        "packagePatterns": ["*"],
+        "enabled": false
+      },
+      {
+        "groupName": "docker-dependencies",
+        "matchManagers": ["dockerfile"],
+        "matchUpdateTypes": ["minor", "patch", "major"],
+        "matchPackageNames": [
+          "k8s.gcr.io/kustomize/kustomize",
+          "alpine/helm",
+          "alpine",
+          "memcached",
+          "golang"
+        ]
```

Comment:
I'd suggest breaking this into separate groups per base image, so that we get one PR for Alpine updates, and another PR for Golang updates etc. 

Otherwise, they'll all be grouped together, which may not be desirable (eg. we only want to update Alpine or Golang right now, but not both).

Reply:
Hello, I apologize for my delayed response. I wanted to let you know that we have received approval from the security team, and I have submitted a request to install the Renovate GitHub app on Mimir. The main reason for this is to streamline the process for updating the Dockerfile located at https://github.com/grafana/mimir/blob/main/mimir-build-image/Dockerfile. By doing this, we can reduce the number of pull requests required for updating this file. Each update currently incurs a significant one-hour build and image push, so I'm considering separating only the "golang" component from the others. This change would result in us having a total of maximum 3 builds instead of the potential maximum of 15 or a minimum of 5 builds (calculated as 5 + 4 + 3 + 2 + 1 since each merge and rebase could trigger an additional build and push). What are your thoughts on this approach?

Reply:
How often are all of these base images updated around the same time? I'd err on the side of more builds and splitting the PRs apart.

Reply:
I have seen it when we activate dependabot only. If it is easier to start with, I can set them separately and we regroup when the case showed up again.